pkg/parser/parser.go

```diff
@@ -0,0 +1,22 @@
+package parser
+
+import (
+	"fmt"
+	"github.com/Joffref/genz/internal/parser"
+	"github.com/Joffref/genz/internal/utils"
+	"github.com/Joffref/genz/pkg/models"
+)
+
+// Parse parses a type from a package
+// Note: patterns is a list of patterns to match packages (e.g . ; ./... ; github.com/Joffref/genz/...)
+func Parse(typeName string, patterns []string, buildTags ...string) (models.ParsedElement, error) {
+	pkg := utils.LoadPackage(patterns, buildTags)
+	if pkg == nil {
+		return models.ParsedElement{}, fmt.Errorf("could not load package")
+	}
+	element, err := parser.Parser(pkg, typeName)
```

Comment:
I still believe we should only expose `parser.Parser(pkg, typeName)`, because all the other code can be performed by the user and **we have no value added**.
Plus, in some cases, packages are loaded by dir path, or by another part of user's program